dataextractor.py

```diff
@@ -0,0 +1,300 @@
+
+    # Save the molecule title
+    rdmol.SetProp("_Name", oemol.GetTitle())
+
+    # Cleanup the rdmol
+    # Note I copied UpdatePropertyCache and GetSSSR from Shuzhe's code to convert oemol to rdmol here:
+    rdmol.UpdatePropertyCache(strict=False)
+    Chem.GetSSSR(rdmol)
+    # I added AssignStereochemistry which takes the directions of the bond set
+    # and assigns the stereochemistry tags on the double bonds
+    Chem.AssignStereochemistry(rdmol, force=False)
+
+    print("Final Molecule: ", Chem.MolToSmiles(Chem.RemoveHs(rdmol), isomericSmiles=True))
+    return rdmol.GetMol()
+
+
+def make_molname_df(directory,ffdirectorylist):
+    """This is the make_molname_df function. It finds the names of mol2 files
+    that are present in all forcefield-minimized directories in the directory.
+    Its output is a pandas dataframe. 
+    """
+    dname = {}
+    # for every forcefield, create dataframe of molnames
+    for ffdirect in ffdirectorylist:
+        tempdirectory = '%s/%s/' % (directory, ffdirect)
+        list_of_mol2 = sorted(os.listdir(tempdirectory))
+        list_of_molnames = list()
+        for mol2 in list_of_mol2:
+            molName = mol2.split('.')[0]
+            list_of_molnames.append(molName)
+        tempnamedf = pd.DataFrame(np.array(list_of_molnames))
+        dname["%s" % ffdirect] = tempnamedf
+    # merges all dataframes generated above into one dataframe
+    # only keeps molecules present in all dataframes
+    twonamedf = dname[ffdirectorylist[0]].merge(dname[ffdirectorylist[1]])
+    for i in range(2, len(ffdirectorylist)):
+        threenamedf = dname[ffdirectorylist[i]].merge(twonamedf)
+    threenamedf = threenamedf.rename({0: "MolNames"},axis='columns')
+    return threenamedf
+
+
+def all_info_df(ffdirectorylist,threenamedf):
+    """
+    This is the all_info_df function. It takes in the list of forcefields,
+    as well as the dataframe of all molecule names, and runs TFD and Tanimoto
+    Combo on all molecules. Its output is a dataframe of all this data. 
+    """
+    #Creating empty dictionaries that TFD and TANI scores will go in later,
+    #as well as a heavyatomlist for putting heavy atoms in
+    heavyatomlist = list()
+    TFDdict = {}
+    TANIdict = {}
+    #creates combinations of forcefields and puts them into dictionaries
+    for i,j in list(itertools.combinations(ffdirectorylist,2)):
+        TFDdict['%s %s' % (i, j)] = {}
+        TANIdict['%s %s' % (i, j)] = {}
+    #generates all the data
+    for molname in threenamedf['MolNames']:
+        print(molname)
+        mol_file = '%s' % molname + '.mol2'
+        refmolin = oechem.oemolistream('%s/%s/%s' % (directory,ffdirectorylist[0],mol_file))
+        refmolhev = oechem.OEGraphMol()
+        oechem.OEReadMolecule(refmolin,refmolhev)
+        #tries to get heavyatom count -- if it can't, returns -1. 
+        try:
+            heavyvalue = oechem.OECount(refmolhev, oechem.OEIsHeavy())
+        except (Exception, ValueError):
```

Comment:
can probably take out the try and except 

can put check when reading in the molecule -line 227 -- if not